### PR TITLE
Added user timings hits and unit tests.

### PR DESCRIPTION
**Pull request by sriverag**
Added timing() function to ReactGA object and unit tests. 

Reference
https://developers.google.com/analytics/devguides/collection/analyticsjs/user-timings#overview

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -153,6 +153,45 @@ var ReactGA = {
   },
 
   /**
+   * timing:
```
Comment (sriverag) [inline src/index.js]: Changes start here.

File: src/index.js [hunk 2]
```diff
@@ -153,6 +153,45 @@ var ReactGA = {
...
+      if (!args || !args.category || !args.variable
+          || !args.value || typeof args.value !== 'number') {
+        warn('args.category, args.variable ' +
+            'AND args.value are required in timing() ' +
+            'AND args.value has to be a number');
+        return;
+      }
+
+      //Required Fields
+      var fieldObject = {
```
Comment (sriverag) [inline src/index.js]: User timings documentation with fields object structure https://developers.google.com/analytics/devguides/collection/analyticsjs/user-timings#overview

File: test/index.test.js [hunk 3]
```diff
@@ -270,10 +270,105 @@ describe('react-ga', function () {
...
+          hitType: 'exception'
+        }]
+      ]);
+    });
+  });
+
+  /**
+   * timing()
+   */
+
```
Comment (sriverag) [inline test/index.test.js]: Unit tests changes start here

**Comment by yangshun** [comment]
Nice work @sriverag ! But we prefer that you follow the existing style and not change the indentation. The current indentation is meant to be indented at the where the object starts and not all the way to the left.

**Comment by sriverag** [comment]
Thanks @yangshun . I'll fix it and commit with the proper indentation.

**Comment by sriverag** [comment]
Let me know if you find something else.

File: test/index.test.js [hunk 4]
```diff
@@ -359,24 +457,24 @@ describe('react-ga', function () {
       ga.initialize('foo');
       ga.event({ category: 'Test', action: 'Send Test', label: 'Test Label Value' });
       getGaCalls().should.eql([['create', 'foo', 'auto'],
-                                ['send', { eventAction: 'Send Test',
-                                            eventCategory: 'Test',
-                                            eventLabel: 'Test Label Value',
-                                            hitType: 'event'
-                                          }]
-                                ]);
+        ['send', { eventAction: 'Send Test',
```
Comment (yangshun) [inline test/index.test.js]: You missed out on this.

**Comment by yangshun** [comment]
Could you also update the README for this new function? (:

**Comment by sriverag** [comment]
README.md updated!

**Comment by yangshun** [comment]
Great! Lastly, could you squash this PR into a single commit? They're ready to be merged (:

**Comment by sriverag** [comment]
Commits were squashed successfully!

**Comment by yangshun** [comment]
@sriverag I don't think so, this PR still says 9 commits. You'll have to force push to this branch in order for them to be squashed.

**Comment by yangshun** [comment]
It's ok, I realize squashing and merging was available for this repository. Thanks!

**Pull request merged**